### ISSUE TITLE
Improve error message for invalid path dependencies within git dependencies

### ISSUE THREAD

**Issue by NickeZ** [rust-lang/cargo#7784]
**Problem**
I add a git dependency on "noise-rust-crypto" which is part of the "noise-rust" repository and cargo doesn't find it.

```
$ cargo update
    Updating git repository `https://github.com/sopium/noise-rust`
error: failed to load source for a dependency on `noise-rust-crypto`

Caused by:
  Unable to update https://github.com/sopium/noise-rust

Caused by:
  Could not find `Cargo.toml` in `/home/niklas/.cargo/git/checkouts/noise-rust-6131949caf0d2152/dad7f8b/noise`
```

**Steps**

1. `echo "noise-rust-crypto = {git = "https://github.com/sopium/noise-rust"}" >> Cargo.toml`
2. cargo update
3.

**Possible Solution(s)**

**Notes**

Output of `cargo version`:

```
$ cargo --version
cargo 1.40.0 (bc8e4c8be 2019-11-22)
```

**Comment by ehuss** [comment]
It's a poor error message.  The issue is that [this line](https://github.com/sopium/noise-rust/blob/dad7f8b49a9643ca9397765e22a1ea47626785a4/noise-ring/Cargo.toml#L18) references a directory that doesn't exist.

**Comment by NickeZ** [comment]
Ok so it actually breaks when it fails instead of continue to search. That sounds brittle. I'll make a pr to the repo. Thanks

**Comment by ehuss** [comment]
Yea, it is supposed to ignore errors like this. It's just an oversight that it doesn't in this particular case (`path` dependencies require a special code path).

**Comment by NickeZ** [comment]
I tried to update the issue name to something actionable. Thanks!

**Comment by ehuss** [comment]
I think this is now resolved by #7784.  Bad paths in the git repo are now ignored (unless the package itself is missing).

**Comment by aleksator** [comment]
> I think this is now resolved by #7784. Bad paths in the git repo are now ignored (unless the package itself is missing).

The link in the message points to this issue. Recursion!
